event-streams/src/event-streams-v1.0-wd01.md

```diff
@@ -0,0 +1,547 @@
+property](#211-event-streams-partition-link-property). If the requested
+partition does not exist, attaching the link MUST be rejected by the event log
+node. The event log node MAY assign a partition when none has been requested. It
+MUST NOT assign a partition other than the one that has been requested.
+
+If no partition is assigned, the default behavior MUST be for a `send` link to
+be partition agnostic and for any partition-related hints to flow via delivery
+annotations, which are defined further below in this section.
+
+A consumer's `receive` link being attached to an event log node MAY request
+association with a specific partition using the the [`event-streams-partition`
+link property](#211-event-streams-partition-link-property). If the requested
+partition does not exist, attaching the link MUST fail. If the event log is
+partitioned and no association has been requested, the event log node MUST
+associate a partition with the link.
+
+For both `send` and `receive` links, the effective associated partition MUST be
+expressed with the [`event-streams-partition` link
+property](#211-event-streams-partition-link-property) set by the event log node.
+
+### 2.1.1 event-streams-partition link property
+
+The `event-streams-partition` link property is a `symbol` value and describes
+the partition with which the link is associated. When set by the producer or
+consumer, the property value is a request for association with the given
+partition. When set by the event log node, the value is the effective
+association.
+
+When not set for a producer 'send' link, the link is partition-agnostic.
+
+### 2.1.2 event-streams-partition delivery annotation
+
+The `event-streams-partition` delivery annotation is a `symbol` value and
+describes the partition to which the message carrying it MUST be routed or from
+which partition is has been retrieved.
+
+If the link over which a message is routed is not partition agnostic with or if
+the link is not associated with the exact same partition, the transfer MUST be
+rejected.
+
+If the event log is partitioned, the delivery annotation MUST be added by the
+event log node before delivery to consumers. A consumer SHOULD strip the
+annotation before forwarding it.
+
+### 2.1.3 event-streams-group-key message annotation
+
+The `event-streams-key` message annotation is a `string` value and
+is a grouping key that identifies events that are related and SHOULD therefore
+be treated as a group for partitioning purposes.
+
+This is an annotation and therefore differs from the core message property
+`group-id` by allowing overrides as messages get routed through multiple
+partitioned intermediaries where partitioning concerns might vary.
+
+When set by the producer and used over a partition agnostic link, the value of
+the annotation MAY be used by the event log to associate the message with a
+partition. If the key is used in this way, the event log SHOULD associate the
```

Comment:
but now it seems to say that logic of destination partition is offloaded on the event log node. Always compared to Kafka, if you use a key, it's the producer to process the key and getting out a destination partition while in this case the producer seems to be allowed to specify a key, and the logic "key to partition" is on the event log node.